src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+ */
+var userChangeDispatch = function () {
+
+    return (inFields, action) => {
+        if (!inFields) {
+            return fieldInit();
+        } else {
+            switch (action.type) {
+                // update the size field in case tab selection is changed
+                case FieldGroupCntlr.VALUE_CHANGE:
+                    if (action.payload.fieldKey === 'targettry') {
+
+                        break;
+
+                    }
+                    const valP = inFields.project.value;
+                    let valC = inFields.catalog.value;
+                    const optList = getSubProjectOptions(catmaster, valP);
+                    if (action.payload.fieldKey === 'project') {
+                        valC = optList[0].value;//If project has changed, initiialise to the first subproject found
+                        inFields = updateMerge(inFields, 'catalog', {
+                            value: valC
+                        });
+                    }
+                    inFields = updateMerge(inFields, 'tableview', {
+                        selProj: valP,
+                        selCat: valC
+                    });
+
+                    // Reinit the table and catalog value:
+                    const catTable = getCatalogOptions(catmaster, valP, valC).option;
+
+                    let sizeFactor = 1;
+                    const method = get(inFields, 'spatial.value', SpatialMethod.Cone.value);
+                    if (method === SpatialMethod.Box.value) {
+                        sizeFactor = 2;
+                    }
+
+                    //No need to act on other fields if user change user wpt or spatial method
+                    if (action.payload.fieldKey === 'UserTargetWorldPt'
+                        || action.payload.fieldKey === 'conesize') {
+
+                        break;
+
+                    }
+                    let idx = get(inFields, 'cattable.indexClicked', 0);//Get current value
+                    if (action.payload.fieldKey === 'project'
+                        || action.payload.fieldKey === 'catalog') {
+                        idx = 0; // reset to first item of the catalog list
+                    }
+
+                    // User clicked on the table to select a catalog and needs to propagte the index in order
+                    // to get the item highlighted
+                    if (action.payload.fieldKey === 'cattable') {
+                        idx = catTable.findIndex((e) => {
+                            return e.value === action.payload.value;
+                        });
+
+                    }
+                    const radius = parseFloat(catTable[idx].cat[7]);
+
+                    inFields = updateMerge(inFields, 'cattable', {
+                        groupKey: gkey,
+                        fieldKey: 'cattable',
+                        indexClicked: idx,
+                        value: catTable[idx].value
+                    });
```

Comment:
yes, you are right.

On 5/16/16 3:39 PM, loitly wrote:
>
> In src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx 
> <https://github.com/Caltech-IPAC/firefly/pull/85#discussion_r63438167>:
>
> > +                    // User clicked on the table to select a catalog and needs to propagte the index in order
> > +                    // to get the item highlighted
> > +                    if (action.payload.fieldKey === 'cattable') {
> > +                        idx = catTable.findIndex((e) => {
> > +                            return e.value === action.payload.value;
> > +                        });
> > +
> > +                    }
> > +                    const radius = parseFloat(catTable[idx].cat[7]);
> > +
> > +                    inFields = updateMerge(inFields, 'cattable', {
> > +                        groupKey: gkey,
> > +                        fieldKey: 'cattable',
> > +                        indexClicked: idx,
> > +                        value: catTable[idx].value
> > +                    });
>
> because this is a merge, you do not need to set props that did not 
> changed.
> i am sure groupKey and fieldKey are not changed. by merging in only 
> the changes, it'll be clearer what it is you are changing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/85/files/8bdfd737fd8631b45d10aff1ebb2742bee675e9e#r63438167>
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489